accepted/future-releases/constructor-tearoffs/faq.md

```diff
@@ -0,0 +1,262 @@
+f((a<b, c>) - d)
+```
+
+**Q:** Can I call a static method from `List` on `List<int>`.
+
+**A:** No. You can write `List.copyRange` but not `List<int>.copyRange`. This is a grammar based restriction. Even if you have a type alias like `typedef Stupid<X> = int;`, where `Stupid<void>` just means `int`, you can't do `Stupid<void>.parseInt`. The only thing you can access through an instantiated type literal is constructors. (So `Stupid<void>.fromEnvironment` is valid, just please don't do it.)
+
+**Q:** What if I want to call an extension method defined on `Type` on a `List<int>`?
+
+**A:** Then you need parentheses, just like now. Writing `List<int>.filled` *only works for constructors*. If you write `.something` after an explicitly instantiated type literal, it tries to do a constructor lookup. If you write `.something` after a raw type literal, it tries to do a static member or constructor lookup (that's what you can do now). If you want to call instance or extension members on `Type`, whether an extension or just `.toString()`, you need to write `(List<int>).toString()`.
+
+**Q:** If I can instantiate a generic function, and `List.filled` is a generic function tear-off, why is `List.filled<int>` invalid?
+
+**A:** Because we say so. We could make it mean `(List.filled)<int>`, but we prefer to reserve the syntax for if/when we add proper generic constructors. And you *can* write the parentheses, just please don't. Prefer `List<int>.filled` if that's what you mean&mdash;and if it's not what you mean, you probably can't yet do what you mean.
+
+**Q:** So where can I write an explicit type argument instantiation.
+
+**A:** In short, after any expression which can actually use type arguments. The meaning depends on what generic thing is being instantiated.
+
+- If the expression denotes a generic class, mixin, or type alias, it just supplies type parameters to the type, e.g., `Type t = List<int>;` If the type is then followed by a constructor name, then it's a constructor tear-off, e.g., `var f = List<int>.filled;`. Otherwise it's a `Type` literal.
+- If the expression is a generic function, it instantiates it, e.g., `var f = [1.0].map<int>;`.
+- If the expression is a callable object with a generic `call` method, it instantiates the `call` method, e.g., if `foo.call` is a generic method, then `var f = foo<int>;` is equivalent to `var f = foo.call<int>;`.
+
+In all other cases, it's an error. You can only instantiate something which is generic.
+
+**Q:** Can I do an explicit instantiation on a `dynamic` value?
+
+**A:** No. We do allow you to *invoke* a `dynamic` value *v* as <code>*v*\<typeArgs>(args)</code>, but we do *not* allow <code>*v*\<typeArgs></code> as a dynamic instantiation. It's not one of the cases above because they all require or imply having a non-`dynamic` static type. (Same for expressions with static type `Function` or `Never`). In short, we need to know the static type of the type arguments before we will try to instantiate them.
```

Comment:
Maybe change 'having a non-`dynamic` static type ...' to 'having a static type that cannot be `dynamic` (and also cannot be `Function`, or `Never`)'.